opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -1362,73 +1362,117 @@ template cpgrid::Entity<3> Dune::createEntity(const CpGrid&, int, bool);
 template cpgrid::Entity<1> Dune::createEntity(const CpGrid&, int, bool); // needed in distribution_test.cpp 
 
 
-void CpGrid::addLgrUpdateLeafView(const std::array<int,3>& cells_per_dim, const std::array<int,3>& startIJK, const std::array<int,3>& endIJK)
+void CpGrid::addLgrUpdateLeafView(std::array<int,3> cells_per_dim, std::array<int,3> startIJK, std::array<int,3> endIJK)
+{
+    const std::vector<std::array<int,3>> cells_per_dim_vec{{cells_per_dim}};
+    const std::vector<std::array<int,3>> startIJK_vec{{startIJK}};
+    const std::vector<std::array<int,3>> endIJK_vec{{endIJK}};
+    
+    this -> addLgrsUpdateLeafView(cells_per_dim_vec, startIJK_vec, endIJK_vec);
+}
+
+
+void CpGrid::addLgrsUpdateLeafView(const std::vector<std::array<int,3>>& cells_per_dim_vec,
+                                   const std::vector<std::array<int,3>>& startIJK_vec,
+                                   const std::vector<std::array<int,3>>& endIJK_vec)
 {
     if (!distributed_data_.empty()){
-        if (comm().rank()==0)
-        {
-            OPM_THROW(std::logic_error, "Adding LGRs to a distributed grid is not supported, yet.");
-        }
-        else
-        {
-            OPM_THROW_NOLOG(std::logic_error, "Adding LGRs to a distributed grid is not supported, yet.");
+        OPM_THROW(std::logic_error, "Grid has been distributed. Cannot create LGRs.");
+    }
+    if (startIJK_vec.size() > 0){
+        if (!(*data_[0]).disjointPatches(startIJK_vec, endIJK_vec)){
+        OPM_THROW(std::logic_error, "Patches are not disjoint.");
```

Comment:
Here we should use the same approach as above:
```
if (startIJK_vec.size() > 0) && !(*data_[0]).disjointPatches(startIJK_vec, endIJK_vec){
        if (comm().rank()==0)
        {
            OPM_THROW(std::logic_error, "LGRs are not disjoint.");
        }
        else
        {
            OPM_THROW_NOLOG(std::logic_error, "LGRs are not disjoint.");
        }
}
```

Reply:
Added!